format/format.go

```diff
@@ -359,10 +381,24 @@ func (w *writer) writeComments(comments []*ast.Comment) {
 }
 
 func (w *writer) writeRules(rules []*ast.Rule, o fmtOpts, comments []*ast.Comment) []*ast.Comment {
+	existingAnnotations, _ := ast.ParseAnnotations(comments)
+
 	for _, rule := range rules {
+
+		if len(existingAnnotations) == 0 {
+			for _, an := range rule.Annotations {
+				w.blankLine()
```

Comment:
This will sometimes add an additional blank line. From this part in the code it's not possible to figure out what the previous line is unless we read the buffer.